src/App.tsx

```diff
@@ -1,56 +1,59 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { PeopleSearchBar } from './components/PeopleSearchBar';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Person | null>(null);
+  const [onFocus, setOnFocus] = useState(false);
 
-  return (
-    <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
-
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
-          <input
-            type="text"
-            placeholder="Enter a part of the name"
-            className="input"
-          />
-        </div>
-
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
+  const filteredPeople = useMemo(() => {
+    if (onFocus && !query.trim()) {
+      return peopleFromServer;
+    }
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [peopleFromServer, appliedQuery, onFocus, query]);
 
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
+  const onFocusHandler = useCallback(() => {
+    setOnFocus(true);
+  }, []);
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
+  const onBlurHandler = useCallback(() => {
+    setOnFocus(false);
```

Comment:
why does onBlurHandler handles focus?

Reply:
I need it to know if input is focused. And switch focus flag.